### PR TITLE
Spawn protection for placing blocks. Fixes VANILLA-159

### PR DESCRIPTION
**Pull request by greatman**
Signed-off-by: William Lebel lebelwilliam@gmail.com

### CODE REVIEW

File: src/main/java/org/spout/vanilla/protocol/handler/PlayerBlockPlacementMessageHandler.java [hunk 1]
```diff
@@ -193,7 +199,17 @@ public void handleServer(Session session, PlayerBlockPlacementMessage message) {
...
 					}
 				}
-
+				
+				//Check if the player can place the block.
+				Collection<Protection> protections = Spout.getEngine().getServiceManager().getRegistration(ProtectionService.class).getProvider().getAllProtections(alterBlock.getPosition());
+				for (Protection p : protections) {
+					if (p.contains(alterBlock.getPosition()) && !((VanillaPlayerController)player.getController()).isOp()) {
+						undoPlacement(player, clickedBlock, alterBlock);
+						player.sendMessage(ChatStyle.DARK_RED, "This area is a protected spawn point!");
```
Comment (ZNickq) [inline src/main/java/org/spout/vanilla/protocol/handler/PlayerBlockPlacementMessageHandler.java:208]: Doesn't this check all the protections, not just the spawn point?

Comment (greatman) [inline src/main/java/org/spout/vanilla/protocol/handler/PlayerBlockPlacementMessageHandler.java:208]: Well I copied the same code as in the PlayerDiggingMsg.

Comment (ZNickq) [inline src/main/java/org/spout/vanilla/protocol/handler/PlayerBlockPlacementMessageHandler.java:208]: @NinjaZidane wrote that code i believe...

Comment (Zidane) [inline src/main/java/org/spout/vanilla/protocol/handler/PlayerBlockPlacementMessageHandler.java:208]: It needs to be re-written with API in mind.

**Pull request merged**